incubator/types-core/src/main/java/io/aklivity/zilla/runtime/types/core/StringValidatorHandler.java

```diff
@@ -77,31 +73,19 @@ public boolean validate(
                     }
                 }
                 index += charByteCount;
```

Comment:
Feels like we are double counting the char bytes when fragmented, since `index++` will happen for each pending byte and `index += charByteCount` will happen for complete chars.

Perhaps we can simplify this algorithm to work from the perspective of `pendingBytes` always, and handle complete chars as a special case, where we need to reinitialize the number of `pendingBytes` from the first character byte?